src/prosemirrorSetup.ts

```diff
@@ -1,28 +1,20 @@
-// Mix the nodes from prosemirror-schema-list into the basic schema to
+import type { Node, Schema } from "prosemirror-model";
+import { DOMParser, DOMSerializer } from "prosemirror-model";
 
-import type OrderedMap from "orderedmap";
-import type { Node, NodeSpec } from "prosemirror-model";
-import { DOMParser, DOMSerializer, Schema } from "prosemirror-model";
-import { schema } from "prosemirror-schema-basic";
-import { addEmbedNode } from "./embed";
-
-// create a schema with list support.
-export const mySchema = new Schema({
-  nodes: addEmbedNode(schema.spec.nodes as OrderedMap<NodeSpec>),
-  marks: schema.spec.marks,
-});
-
-const parser = DOMParser.fromSchema(mySchema);
-const serializer = DOMSerializer.fromSchema(mySchema);
+export const createParsers = <S extends Schema>(schema: S) => {
+  const parser = DOMParser.fromSchema(schema);
```

Comment:
Tiny one, why do we need this Type Parameter?

Reply:
It guarantees that the parsers and serialisers that are produced by this function are typed to accept that schema – the type def for DOMParser.fromSchema will infer the schema type from its argument. If we didn't, it'd be typed by default to `Schema<any>`.

To be honest, this isn't necessarily all that important, as the Prosemirror types for `Schema` are a bit anemic anyway – it's more of a nice to have.